contrib/registry.py

```diff
@@ -76,3 +66,104 @@ def model(*models_factories: Factory) -> Union[Factory, List[Factory]]:
             :returns: single model factory or list of them
         """
         return Registry._inner_register("model", *models_factories)
+
+    @staticmethod
+    def module(*modules_factories: Factory) -> Union[Factory, List[Factory]]:
+        """Add module type or factory method to global
+            module list to make it available in config
+            Can be called or used as decorator
+            :param: modules_factories
+                Required module factory (method or type)
+            :returns: single module factory or list of them
+        """
+        return Registry._inner_register("module", *modules_factories)
+
+    @staticmethod
+    def optimizer(
+            *optimizer_factories: Factory
+    ) -> Union[Factory, List[Factory]]:
+        """Add optimizer type or factory method to global
+            optimizer list to make it available in config
+            Can be called or used as decorator
+            :param: optimizer_factories
+                Required optimizer factory (method or type)
+            :returns: single optimizer factory or list of them
+        """
+        return Registry._inner_register("optimizer", *optimizer_factories)
+
+    @staticmethod
+    def name2nn(name):
+        if name is None:
+            return None
+        elif isinstance(name, nn.Module):
+            return name
+        elif isinstance(name, str):
+            return _REGISTERS["module"][name]
+        else:
+            return name
+
+    @staticmethod
+    def create_callback(callback=None, **callback_params):
```

Comment:
Maybe rename these methods?
It's a little confusing, that registry has method `callback` and `create_callback`

also for `optimizers`/`model` ...

Reply:
i propose `get_callback`